code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -87,13 +93,38 @@
 	update_icon()
 	COMPILE_OVERLAYS(src)
 
-/obj/structure/closet/crate/get_door_transform(azimuth_angle, polar_angle, radius_cr)
+/obj/structure/closet/crate/get_door_transform(crateanim_1, crateanim_2)
 		var/matrix/M = matrix()
 		M.Translate(0, -door_hinge)
-		M.Multiply(matrix(1, -sin(polar_angle)*sin(azimuth_angle)* radius_cr, 0, 0, radius_cr*cos(azimuth_angle)*sin(polar_angle), 0))
+		M.Multiply(matrix(1, crateanim_1, 0, 0, crateanim_2, 0))
 		M.Translate(0, door_hinge)
 		return M
 
+/*The animation_list proc writes into two static lists first animation_math where it creates a new list inside for all the values needed for the crate animation.
+The second list is animation_index this list is an associative list where special identify values are saved that are used to determine if there is the need for another list in animation_math or if
+the current lists are sufficent and it is used to get the right index for animation_math for any possible type of crate.
+Next is anindex this variable is just used to save the index for a crate so that you only need to search for the index once.
+The last one is the for loop it simply uses all given indexes to save the needed values to the lists inside animation_math if a new list is required*/
+/obj/structure/closet/crate/proc/animation_list() //pre calculates a list of values for the crate animation cause byond not like math
+	if(animation_index == null) //checks if there is already a list for animation_index if not makes a new list also includes animation_math cause why not honestly
+		animation_index = new/list()
+		animation_math = new/list()
+	anindex = animation_index.Find("[door_anim_time]-[door_anim_angle]-[open_angle]-[radius_2]") //saves index to anindex
+	if(animation_index.Find("[door_anim_time]-[door_anim_angle]-[open_angle]-[radius_2]") == 0)
+		var/num_steps_1 = door_anim_time / world.tick_lag
+		animation_index.len = length(animation_index)+1 //increases list size fo animation index
+		animation_index[length(animation_index)] = "[door_anim_time]-[door_anim_angle]-[open_angle]-[radius_2]" //saves the unique value to the asssociative list
+		anindex = animation_index.Find("[door_anim_time]-[door_anim_angle]-[open_angle]-[radius_2]")
+		animation_math.len = anindex //also just increases the list size
+		animation_math[anindex] = new/list(num_steps_1*2+1) //saves another list into animation_math on the index that just got added
+		for(var/I in 0 to num_steps_1) //loop to save the animation values into the lists
+			var/angle_1 =  I==0 ? 0 : door_anim_angle * (I/num_steps_1)
+			var/polar_angle = abs(arcsin(cos(angle_1)))
+			var/azimuth_angle = angle_1 >= 90 ? open_angle : 0
+			var/radius_cr = angle_1 >= 90 ? radius_2 : 1
+			animation_math[anindex][I+1] = -sin(polar_angle)*sin(azimuth_angle)*radius_cr
+			animation_math[anindex][num_steps_1+I+1] = radius_cr*cos(azimuth_angle)*sin(polar_angle)
```

Comment:
```suggestion
	var/num_steps_1 = door_anim_time / world.tick_lag
	var/list/new_animation_math_sublist[num_steps_1 * 2 + 1]
	animation_math["[door_anim_squish]-[door_anim_angle]-[door_hinge]-[door_anim_time]-[radius_2]-[open_angle]"] = new_animation_math_sublist
	for(var/I in 0 to num_steps_1) //loop to save the animation values into the lists
		var/angle_1 = I == 0 ? 0 : door_anim_angle * (I / num_steps_1)
		var/polar_angle = abs(arcsin(cos(angle_1)))
		var/azimuth_angle = angle_1 >= 90 ? open_angle : 0
		var/radius_cr = angle_1 >= 90 ? radius_2 : 1
		new_animation_math_sublist[I+1] = -sin(polar_angle) * sin(azimuth_angle) * radius_cr
		new_animation_math_sublist[num_steps_1+I+1] = cos(azimuth_angle) * sin(polar_angle) * radius_cr
```

At the suggested rewrite, the proc *will* do the pre-calculation and save it to the `animation_math`.

Reply:
sadly does not seem to work only ends in a runtime with bad index i would love to save the index directly into the list index but i do not know if dm actually supports that at all reason why i went with that big workaround anyways

Reply:
so yes this should work like in http://www.byond.com/docs/ref/#/list/associations but it does not for some reason no idea if list associations like that are broken at the current byond version

Reply:
Not sure assoc lists are broken in recent versions, but the currently recommended version is 1536